libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs

```diff
@@ -1253,12 +1253,12 @@ fn issue_10118() {
         model User {
           id         String    @id @default(cuid()) @db.Char(30)
           referralId String    @unique @db.Char(30)
-          referral   Referral? @relation("UserToReferral", fields: [referralId], references: [id], map: "User_referralId_fkey")
+          referral   Referral? @relation("UserToReferral", fields: [referralId], references: [id])
```

Comment:
Why are test outcomes changing here?

Reply:
Because I switched that thing from `CompleteInlineRelationWalker<'_>`, which does something weird with foreign key naming, to `InlineRelationWalker<'_>`, which has less magic. As a result, I think this is more correct:

- It's a one-to-one relation with an optional and a required side
- In this case, we put the foreign key on the _required_ side (otherwise you can't enforce the requiredness)
- The previous behaviour was putting the `map` argument on the optional side, which doesn't make sense if the foreign key is on the other side
- The whole test still doesn't make sense because you can't have the `fields` and `references` arguments on both sides in a 1:1 relation